core/templates/pages/exploration-editor-page/translation-tab/state-translation/state-translation.component.ts

```diff
@@ -189,12 +205,19 @@ angular.module('oppia').component('stateTranslation', {
           );
           activeContentId = activeContent.contentId;
           if (activeContent instanceof SubtitledUnicode) {
-            activeDataFormat = WRITTEN_TRANSLATION_TYPE_UNICODE;
+            activeDataFormat = TRANSLATION_DATA_FORMAT_UNICODE;
           }
+        } else if (tabId === $scope.TAB_ID_RULE_INPUTS) {
+          const {
+            rule, inputName, contentId
+          } = $scope.interactionRuleTranslatableContents[0];
+          activeContentId = contentId;
+          var inputType = rule.inputTypes[inputName];
+          activeDataFormat = RULE_INPUT_TYPES_TO_DATA_FORMATS[inputType];
```

Comment:
Doesn't `RULE_INPUT_TYPES_TO_DATA_FORMATS` only have `TRANSLATION_DATA_FORMAT_SET_OF_UNICODE_STRING` and `TRANSLATION_DATA_FORMAT_SET_OF_NORMALIZED_STRING` as data formats? Are other data formats not expected in any rule inputs (I thought only text input and set input had 'array' translations)?

Reply:
This is correct and yes, no other data formats are expected in translatable rule inputs at the moment. The two interaction types you mention correspond to the two data formats you mention.

Would you like me to change anything here?

Reply:
Oh, I thought this is a general function. So, what would happen here if someone entered the rules tab for some other interaction? Is this not called in that case?

Reply:
The rules tab in the translation editor would be grayed out, so it would be impossible to enter anyway (at least, that's the theory).

Happy to put more safeguards here, but not sure what to add exactly...?

Reply:
Can you leave a comment mentioning that only 'TextInput' and 'SetInput' have translatable rule types?